implementations/c/ockam/transport/posix_socket/socket_tcp.c

```diff
@@ -296,15 +295,15 @@ ockam_error_t socket_tcp_read(void* ctx, uint8_t* buffer, size_t buffer_size, si
 
 ockam_error_t socket_tcp_write(void* ctx, uint8_t* buffer, size_t buffer_length)
 {
+  // TODO: Check buffer_length can fit in ssize_t?
   ockam_error_t   error       = OCKAM_ERROR_NONE;
   tcp_socket_t*   p_tcp_ctx   = (tcp_socket_t*) ctx;
   posix_socket_t* p_socket    = &p_tcp_ctx->posix_socket;
   uint16_t        send_length = 0;
-  size_t          bytes_sent  = 0;
 
   // Convert from native endianness to network order
   send_length = htons(buffer_length);
-  bytes_sent  = send(p_socket->socket_fd, (void*) &send_length, sizeof(uint16_t), 0);
+  ssize_t bytes_sent = send(p_socket->socket_fd, (void*) &send_length, sizeof(uint16_t), 0);
```

Comment:
is this ever negative?

Reply:
It can be -1

Reply:
if -1 is cast to unsigned, does it saturate at uint16_t.MAX?

Reply:
@mikelodder7 yes, but it should not be problem here.

Reply:
A saturation cast means -1 == uint16_t.MAX, i.e. the if statement will succeed. Does this lead to unexpected behavior?

Reply:
@mikelodder7 sizeof(uint16_t) is 2